gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-scim/src/main/java/io/gravitee/am/gateway/handler/scim/service/impl/UserServiceImpl.java

```diff
@@ -467,4 +484,9 @@ private Completable checkRoles(List<String> roles) {
                 }).ignoreElement();
     }
 
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    private Maybe<PasswordHistory> createPasswordHistory(Domain domain, io.gravitee.am.model.User user, String rawPassword, io.gravitee.am.identityprovider.api.User principal) {
+        return passwordHistoryService
+                .addPasswordToHistory(DOMAIN, domain.getId(), user, rawPassword , principal, domain.getPasswordSettings());
```

Comment:
We are only managing the domain settings, we should use `PasswordSettings.getInstance()`

Reply:
Where is the reference for client/application in the scim user service?

Reply:
There is no difference in term of concept application is the same as client.
Client is used in the GW for legacy reason (and to keep the same naming as the one defined into the OAuth spec)

To convert an application object to a client object, you can do `application.toClient()`;

Reply:
I'm not seeing any application reference either.